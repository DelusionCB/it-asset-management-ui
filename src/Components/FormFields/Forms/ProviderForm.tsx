import React, {useEffect, useState} from 'react';
import {Row} from 'reactstrap';
import {formTypes} from '../../../Types/types.forms';
import {providerDataPropTypes} from '../../../Types/types.directories';
import {providerDefaultValues} from '../../../Types/defaultValues';
import {useTranslation} from 'react-i18next';
import {debouncedCheck, handleProviderChange, handleProviderSelect, handleClear} from '../../../Utils/helpers';
import VisibilitySelector from '../../CustomComponents/VisibilitySelector';
import TextField from '../../CustomComponents/TextField';
import StaticSelect from '../../CustomComponents/StaticSelect';
import SearchableSelect from '../../CustomComponents/SearchableSelect';
import ActionButtons from '../../ActionButtons/ActionButtons';

function ProviderForm ({isDisabled, type, mode, editValues}: formTypes): JSX.Element {
    const [values, setValues] = useState<providerDataPropTypes>(providerDefaultValues)
    const {t} = useTranslation()

    function handleChange (e: React.ChangeEvent<HTMLInputElement>): void {
        handleProviderChange(e, values, setValues);
    }

    function handleSelect (value: {} | [] | undefined | null, id: string): void {
        handleProviderSelect(value, id, values, setValues);
    }

    useEffect(() => {
        if (Object.keys(editValues).length !== 0 && mode === 'edit') {
            setValues(editValues)
        }
    }, [editValues, mode])

    useEffect(() => {
        if (mode !== 'edit') {
            // Invoke debouncedCheck with updated values
            debouncedCheck(values, providerDefaultValues, isDisabled);

            // Cleanup function to cancel debounce on unmount
            return () => {
                debouncedCheck.cancel();
            };
        }
    }, [values]);

    useEffect(() => {
        if (values.visibility === 'hidden' || values.visibility === 'draft') {
            // Clear these fields if visibility is X -value
            handleClear<providerDataPropTypes>(
                values,
                [],
                ['related_services', 'related_applications', 'related_contracts'],
                setValues,
            )
        }
    }, [values.visibility])

    return (
        <>
            <Row>
                <h1>{t(`provider.${mode}`)}</h1>
            </Row>
            <Row>
                <h2>{t('provider.visibility')}</h2>
            </Row>
            <Row>
                <VisibilitySelector
                    value={values.visibility}
                    onChange={(e) => { handleChange(e); }}
                    disabled={false}
                    id='visibility'
                    label={'values.visibility'}
                    invalid={false}
                    options={['draft', 'hidden', 'published']}
                    placeholder={'placeholder.visibility'}
                />
            </Row>
            <Row>
                <h2>{t('provider.description')}</h2>
            </Row>
            <Row>
                <TextField
                    id='name'
                    disabled={false}
                    invalid={false}
                    label={'provider.values.name'}
                    onChange={(e) => { handleChange(e); }}
                    placeholder={'placeholder.name'}
                    value={values.name}
                    type='text'
                />
            </Row>
            <Row>
                <TextField
                    id={'description'}
                    disabled={false}
                    invalid={false}
                    type='textarea'
                    label={'provider.values.description'}
                    onChange={(e) => { handleChange(e) }}
                    placeholder={'placeholder.description'}
                    value={values.description}
                />
            </Row>
            <Row>
                <TextField
                    id={'business_id'}
                    disabled={false}
                    invalid={false}
                    type='text'
                    label={'provider.values.business_id'}
                    onChange={(e) => { handleChange(e) }}
                    placeholder={'placeholder.business_id'}
                    value={values.business_id}
                />
            </Row>
            <Row>
                <StaticSelect
                    id={'provider_type'}
                    disabled={false}
                    invalid={false}
                    label={'provider.values.provider_type'}
                    onChange={(e) => { handleChange(e) }}
                    placeholder={'placeholder.provider_type'}
                    value={values.provider_type}
                    options={['framework', 'inhouse']}
                />
            </Row>
            <Row>
                <h2>{t('provider.contacts')}</h2>
            </Row>
            <Row>
                <TextField
                    id={'full_address'}
                    disabled={false}
                    invalid={false}
                    type='text'
                    label={'provider.values.full_address'}
                    onChange={(e) => { handleChange(e) }}
                    placeholder={'placeholder.full_address'}
                    value={values.full_address}
                />
            </Row>
            <Row>
                <TextField
                    id={'switch_phone'}
                    disabled={false}
                    invalid={false}
                    type='text'
                    label={'provider.values.switch_phone'}
                    onChange={(e) => { handleChange(e) }}
                    placeholder={'placeholder.switch_phone'}
                    value={values.switch_phone}
                />
            </Row>
            <Row>
                <TextField
                    id={'general_email'}
                    disabled={false}
                    invalid={false}
                    type='email'
                    label={'provider.values.general_email'}
                    onChange={(e) => { handleChange(e) }}
                    placeholder={'placeholder.general_email'}
                    value={values.general_email}
                />
            </Row>
            <Row>
                <TextField
                    id={'support_phone'}
                    disabled={false}
                    invalid={false}
                    type='text'
                    label={'provider.values.support_phone'}
                    onChange={(e) => { handleChange(e) }}
                    placeholder={'placeholder.support_phone'}
                    value={values.support_phone}
                />
            </Row>
            <Row>
                <TextField
                    id={'support_email'}
                    disabled={false}
                    invalid={false}
                    type='email'
                    label={'provider.values.support_email'}
                    onChange={(e) => { handleChange(e) }}
                    placeholder={'placeholder.support_email'}
                    value={values.support_email}
                />
            </Row>
            <Row>
                <TextField
                    id={'additional_contact'}
                    disabled={false}
                    invalid={false}
                    type='textarea'
                    label={'provider.values.additional_contact'}
                    onChange={(e) => { handleChange(e) }}
                    placeholder={'placeholder.additional_contact'}
                    value={values.additional_contact}
                />
            </Row>
            <Row>
                <h2>{t('provider.dependency')}</h2>
            </Row>
            <Row>
                <SearchableSelect
                    label={'provider.values.related_services'}
                    endpoint='service'
                    id={'related_services'}
                    placeholder={'placeholder.related_services'}
                    disabled={values.visibility === 'hidden' || values.visibility === 'draft'}
                    invalid={false}
                    isMulti={true}
                    value={values.related_services}
                    onChange={(value, id) => { handleSelect(value, id); }}
                />
            </Row>
            <Row>
                <SearchableSelect
                    label={'provider.values.related_applications'}
                    endpoint='application'
                    id={'related_applications'}
                    placeholder={'placeholder.related_applications'}
                    disabled={values.visibility === 'hidden' || values.visibility === 'draft'}
                    invalid={false}
                    isMulti={true}
                    value={values.related_applications}
                    onChange={(value, id) => { handleSelect(value, id); }}
                />
            </Row>
            <Row>
                <SearchableSelect
                    label={'provider.values.related_contracts'}
                    endpoint='contract'
                    id={'related_contracts'}
                    placeholder={'placeholder.related_contracts'}
                    disabled={values.visibility === 'hidden' || values.visibility === 'draft'}
                    invalid={false}
                    isMulti={true}
                    value={values.related_contracts}
                    onChange={(value, id) => { handleSelect(value, id); }}
                />
            </Row>
            <Row>
                <TextField
                    id={'provider_user_contact'}
                    disabled={false}
                    invalid={false}
                    type='text'
                    label={'provider.values.provider_user_contact'}
                    onChange={(e) => { handleChange(e) }}
                    placeholder={'placeholder.provider_user_contact'}
                    value={values.provider_user_contact}
                />
            </Row>
            <Row>
                <h2>{t('provider.extra')}</h2>
            </Row>
            <Row>
                <TextField
                    id={'extra_url'}
                    disabled={false}
                    invalid={false}
                    type='text'
                    label={'provider.values.extra_url'}
                    onChange={(e) => { handleChange(e) }}
                    placeholder={'placeholder.extra_url'}
                    value={values.extra_url}
                />
            </Row>
            <ActionButtons
                values={values}
                type={type}
                mode={mode}
            />
        </>
    );
}

export default ProviderForm;
