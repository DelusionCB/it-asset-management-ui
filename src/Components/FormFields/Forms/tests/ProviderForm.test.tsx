import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom'
import '@testing-library/jest-dom'
import ProviderForm from '../ProviderForm';

describe('<ProviderForm />', () => {
    let renderer: TestRenderer.ReactTestRenderer;

    // Render the component before tests
    beforeAll(() => {
        renderer = TestRenderer.create(
            <BrowserRouter>
                <ProviderForm editValues={{}} mode='create' type='provider' isDisabled={jest.fn()} />
            </BrowserRouter>,
        );
    });
    describe('Snapshot', () => {
        it('renders correctly', () => {
            expect(renderer.toJSON()).toMatchSnapshot();
        })
    })
})
