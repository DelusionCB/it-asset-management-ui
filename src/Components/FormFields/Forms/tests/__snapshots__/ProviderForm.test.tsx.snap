// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProviderForm /> Snapshot renders correctly 1`] = `
[
  <div
    className="row"
  >
    <h1>
      provider.create
    </h1>
  </div>,
  <div
    className="row"
  >
    <h2>
      provider.visibility
    </h2>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="visibility"
        >
          values.visibility
        </label>
        <select
          aria-invalid={false}
          className="form-select"
          disabled={false}
          id="visibility"
          name="visibility"
          onChange={[Function]}
          placeholder="placeholder.visibility"
          value="published"
        >
          <option
            value="draft"
          >
            visibility.draft
          </option>
          <option
            value="hidden"
          >
            visibility.hidden
          </option>
          <option
            value="published"
          >
            visibility.published
          </option>
        </select>
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <h2>
      provider.description
    </h2>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="name"
        >
          provider.values.name
        </label>
        <input
          aria-invalid={false}
          className="form-control"
          disabled={false}
          id="name"
          name="name"
          onChange={[Function]}
          placeholder="placeholder.name"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="description"
        >
          provider.values.description
        </label>
        <textarea
          aria-invalid={false}
          className="form-control"
          disabled={false}
          id="description"
          name="description"
          onChange={[Function]}
          placeholder="placeholder.description"
          value=""
        />
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="business_id"
        >
          provider.values.business_id
        </label>
        <input
          aria-invalid={false}
          className="form-control"
          disabled={false}
          id="business_id"
          name="business_id"
          onChange={[Function]}
          placeholder="placeholder.business_id"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="provider_type"
        >
          provider.values.provider_type
        </label>
        <select
          aria-invalid={false}
          className="form-select"
          disabled={false}
          id="provider_type"
          name="provider_type"
          onChange={[Function]}
          placeholder="placeholder.provider_type"
          value="inhouse"
        >
          <option
            value="framework"
          >
            select.framework
          </option>
          <option
            value="inhouse"
          >
            select.inhouse
          </option>
        </select>
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <h2>
      provider.contacts
    </h2>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="full_address"
        >
          provider.values.full_address
        </label>
        <input
          aria-invalid={false}
          className="form-control"
          disabled={false}
          id="full_address"
          name="full_address"
          onChange={[Function]}
          placeholder="placeholder.full_address"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="switch_phone"
        >
          provider.values.switch_phone
        </label>
        <input
          aria-invalid={false}
          className="form-control"
          disabled={false}
          id="switch_phone"
          name="switch_phone"
          onChange={[Function]}
          placeholder="placeholder.switch_phone"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="general_email"
        >
          provider.values.general_email
        </label>
        <input
          aria-invalid={false}
          className="form-control"
          disabled={false}
          id="general_email"
          name="general_email"
          onChange={[Function]}
          placeholder="placeholder.general_email"
          type="email"
          value=""
        />
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="support_phone"
        >
          provider.values.support_phone
        </label>
        <input
          aria-invalid={false}
          className="form-control"
          disabled={false}
          id="support_phone"
          name="support_phone"
          onChange={[Function]}
          placeholder="placeholder.support_phone"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="support_email"
        >
          provider.values.support_email
        </label>
        <input
          aria-invalid={false}
          className="form-control"
          disabled={false}
          id="support_email"
          name="support_email"
          onChange={[Function]}
          placeholder="placeholder.support_email"
          type="email"
          value=""
        />
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="additional_contact"
        >
          provider.values.additional_contact
        </label>
        <textarea
          aria-invalid={false}
          className="form-control"
          disabled={false}
          id="additional_contact"
          name="additional_contact"
          onChange={[Function]}
          placeholder="placeholder.additional_contact"
          value=""
        />
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <h2>
      provider.dependency
    </h2>
  </div>,
  <div
    className="row"
  >
    <div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="related_services"
        >
          provider.values.related_services
        </label>
        <div
          className=" css-b62m3t-container"
          onKeyDown={[Function]}
        >
          <span
            className="css-1f43avz-a11yText-A11yText"
            id="react-select-2-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
          />
          <div
            className="option-search__control css-13cymwt-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className="option-search__value-container option-search__value-container--is-multi css-1fdsijx-ValueContainer"
            >
              <div
                className="option-search__placeholder css-1jqq78o-placeholder"
                id="react-select-2-placeholder"
              >
                placeholder.related_services
              </div>
              <div
                className="option-search__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-2-placeholder"
                  aria-expanded={false}
                  aria-haspopup={true}
                  aria-invalid={false}
                  aria-label="placeholder.related_services"
                  aria-labelledby="related_services"
                  autoCapitalize="none"
                  autoComplete="off"
                  autoCorrect="off"
                  className="option-search__input"
                  disabled={false}
                  id="related_services"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="combobox"
                  spellCheck="false"
                  style={
                    {
                      "background": 0,
                      "border": 0,
                      "color": "inherit",
                      "font": "inherit",
                      "gridArea": "1 / 2",
                      "label": "input",
                      "margin": 0,
                      "minWidth": "2px",
                      "opacity": 1,
                      "outline": 0,
                      "padding": 0,
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="option-search__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className="option-search__indicator-separator css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className="option-search__indicator option-search__dropdown-indicator css-1xc3v61-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-tj5bde-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div>
            <input
              name="related_services"
              type="hidden"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="related_applications"
        >
          provider.values.related_applications
        </label>
        <div
          className=" css-b62m3t-container"
          onKeyDown={[Function]}
        >
          <span
            className="css-1f43avz-a11yText-A11yText"
            id="react-select-3-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
          />
          <div
            className="option-search__control css-13cymwt-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className="option-search__value-container option-search__value-container--is-multi css-1fdsijx-ValueContainer"
            >
              <div
                className="option-search__placeholder css-1jqq78o-placeholder"
                id="react-select-3-placeholder"
              >
                placeholder.related_applications
              </div>
              <div
                className="option-search__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-3-placeholder"
                  aria-expanded={false}
                  aria-haspopup={true}
                  aria-invalid={false}
                  aria-label="placeholder.related_applications"
                  aria-labelledby="related_applications"
                  autoCapitalize="none"
                  autoComplete="off"
                  autoCorrect="off"
                  className="option-search__input"
                  disabled={false}
                  id="related_applications"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="combobox"
                  spellCheck="false"
                  style={
                    {
                      "background": 0,
                      "border": 0,
                      "color": "inherit",
                      "font": "inherit",
                      "gridArea": "1 / 2",
                      "label": "input",
                      "margin": 0,
                      "minWidth": "2px",
                      "opacity": 1,
                      "outline": 0,
                      "padding": 0,
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="option-search__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className="option-search__indicator-separator css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className="option-search__indicator option-search__dropdown-indicator css-1xc3v61-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-tj5bde-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div>
            <input
              name="related_applications"
              type="hidden"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="related_contracts"
        >
          provider.values.related_contracts
        </label>
        <div
          className=" css-b62m3t-container"
          onKeyDown={[Function]}
        >
          <span
            className="css-1f43avz-a11yText-A11yText"
            id="react-select-4-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
          />
          <div
            className="option-search__control css-13cymwt-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className="option-search__value-container option-search__value-container--is-multi css-1fdsijx-ValueContainer"
            >
              <div
                className="option-search__placeholder css-1jqq78o-placeholder"
                id="react-select-4-placeholder"
              >
                placeholder.related_contracts
              </div>
              <div
                className="option-search__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-4-placeholder"
                  aria-expanded={false}
                  aria-haspopup={true}
                  aria-invalid={false}
                  aria-label="placeholder.related_contracts"
                  aria-labelledby="related_contracts"
                  autoCapitalize="none"
                  autoComplete="off"
                  autoCorrect="off"
                  className="option-search__input"
                  disabled={false}
                  id="related_contracts"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="combobox"
                  spellCheck="false"
                  style={
                    {
                      "background": 0,
                      "border": 0,
                      "color": "inherit",
                      "font": "inherit",
                      "gridArea": "1 / 2",
                      "label": "input",
                      "margin": 0,
                      "minWidth": "2px",
                      "opacity": 1,
                      "outline": 0,
                      "padding": 0,
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="option-search__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className="option-search__indicator-separator css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className="option-search__indicator option-search__dropdown-indicator css-1xc3v61-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-tj5bde-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div>
            <input
              name="related_contracts"
              type="hidden"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="provider_user_contact"
        >
          provider.values.provider_user_contact
        </label>
        <input
          aria-invalid={false}
          className="form-control"
          disabled={false}
          id="provider_user_contact"
          name="provider_user_contact"
          onChange={[Function]}
          placeholder="placeholder.provider_user_contact"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <h2>
      provider.extra
    </h2>
  </div>,
  <div
    className="row"
  >
    <div
      className="textfield col"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="extra_url"
        >
          provider.values.extra_url
        </label>
        <input
          aria-invalid={false}
          className="form-control"
          disabled={false}
          id="extra_url"
          name="extra_url"
          onChange={[Function]}
          placeholder="placeholder.extra_url"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>,
  <div
    className="action-buttons"
  >
    <button
      className="action-button btn btn-secondary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      button.cancel
    </button>
    <button
      className="action-button btn btn-secondary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      button.create
    </button>
  </div>,
]
`;
